.github/scripts/generate_manifests.bash

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+# Script for the initial generation of manifest.yaml files in inputs directories
+
+traverse_dir() {
+  local dir="$1"
+  local contains_subdir=false
+
+  # Loop through the directory contents
+  for file in "$dir"/*; do
+    if [ -d "$file" ]; then
+      contains_subdir=true
+      traverse_dir "$file"
+    fi
+  done
+
+  # If the directory does not contain any subdirectories...
+  if [ "$contains_subdir" = false ]; then
```

Comment:
I believe this conditional is redundant, since to read line 17 the code will already have been false for line 10 (no directories in the parent dir), therefore line 11 would not be run, therefore contains_subdir would be set to the initialisation value (false) every time.

Reply:
But line 10 is in a loop - it's checking that some file in that level of directory is actually a subdirectory. If there is at least one subdirectory, then `contains_subdir` will be `true`.